File: yht0827/1월/1월 10일/방문 길이.java
class Solution {
	static int[] dir = {5, 5};
	static boolean[][][][] map = new boolean[11][11][11][11];
	static int answer = 0;

	public int solution(String dirs) {
		for (String str : dirs.split(""))
			move(str);
		return answer;
	}

	public static void move(String str) {
		int x = dir[0];
		int y = dir[1];

		switch (str) {
			case "U":
				if (++dir[1] > 10) {
					dir[1]--;
					return;
				}
				break;
			case "L":
				if (--dir[0] < 0) {
					dir[0]++;
					return;
				}
				break;
			case "R":
				if (++dir[0] > 10) {
					dir[0]--;
					return;
				}
				break;
			default:
				if (--dir[1] < 0) {
					dir[1]++;
					return;
				}
				break;
		}

		if (!map[x][y][dir[0]][dir[1]] && !map[dir[0]][dir[1]][x][y]) {
			answer++;
			map[x][y][dir[0]][dir[1]] = true;
			map[dir[0]][dir[1]][x][y] = true;
		}
	}
}

File: Eunleelee/2024-01/1월 10일/방문_길이.py
# U : 위, D : 아래, R : 오른쪽, L : 왼쪽
# 처음 걸어본 길의 길이를 구한다
# 좌표를 벗어난 명령은 무시

def solution(dirs):
    width = 0
    length = 0
    answer = 0
    width_dir = [[0 for i in range(10)] for j in range(10)]
    length_dir = [[0 for i in range(10)] for j in range(10)]

    for cmd in dirs:
        if cmd == "U" and length < 5:
            length += 1
        elif cmd == "D" and length > -5:
            length -= 1
        elif cmd == "L" and width > -5:
            width -= 1
        elif cmd == "R" and length < 5:
            width += 1
        else:
            continue

        if (cmd == "U" or cmd == "D") and length_dir[width][length] != 1:
            length_dir[width][length] = 1
            answer += 1
        elif (cmd == "L" or cmd == "R") and width_dir[width][length] != 1:
            width_dir[width][length] = 1
            answer += 1    
          
    return answer

print(solution("UDLRDURL"))